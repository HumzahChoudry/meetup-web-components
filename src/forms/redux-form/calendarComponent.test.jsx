import React from 'react';
import { shallow } from 'enzyme';
import ReduxFormCalendarComponent from './CalendarComponent';

describe('redux-form Calendar Component', function() {
	const attrs = {
		input: {
			id: 'beyonce',
			name: 'halo',
			className: 'beyonce-halo',
			value: new Date(2012, 9, 12),
		},
		meta: {
			error: 'pick a different album',
		},
		onFocus: jest.fn(),
		onBlur: jest.fn(),
		onChange: jest.fn(),
	};

	it('renders a Calendar component with expected attributes from mock data', () => {
		const component = shallow(<ReduxFormCalendarComponent {...attrs} />);
		expect(component).toMatchSnapshot();
	});
});
