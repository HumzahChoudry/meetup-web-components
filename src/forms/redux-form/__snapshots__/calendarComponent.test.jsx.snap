// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux-form Calendar Component renders a Calendar component with expected attributes from mock data 1`] = `
<WithErrorList(CalendarComponent)
  className="beyonce-halo"
  error={null}
  id="beyonce"
  name="halo"
  onBlur={[Function]}
  onFocus={[Function]}
  value={2012-10-12T00:00:00.000Z}
/>
`;
